src/package.ts

```diff
@@ -30,28 +32,147 @@ export interface Package {
   unvalidatedManifest: UnvalidatedPackageManifest;
   validatedManifest: ValidatedPackageManifest;
   changelogPath: string;
+  hasChangesSinceLatestRelease: boolean;
 }
 
 /**
- * Collects information about a package.
+ * Generates possible Git tag names for the root package of a monorepo. The only
+ * tag name in use at this time is "v" + the package version.
  *
- * @param packageDirectoryPath - The path to a package within a project.
+ * @param packageVersion - The version of the package.
+ * @returns An array of possible release tag names.
+ */
+function generateMonorepoRootPackageReleaseTagNames(packageVersion: string) {
+  return [`v${packageVersion}`];
+}
+
+/**
+ * Generates possible Git tag names for the workspace package of a monorepo.
+ * Accounts for changes to `action-publish-release`, which going forward will
+ * generate tags for workspace packages in `PACKAGE_NAME/MAJOR.MINOR.PATCH`, but
+ * in the past merely generated tags for the root package.
+ *
+ * @param packageName - The name of the package.
+ * @param packageVersion - The version of the package.
+ * @param rootPackageVersion - The version of the root package.
+ * @returns An array of possible release tag names.
+ */
+function generateMonorepoWorkspacePackageReleaseTagNames(
+  packageName: string,
+  packageVersion: string,
+  rootPackageVersion: string,
+) {
+  return [`${packageName}@${packageVersion}`, `v${rootPackageVersion}`];
+}
+
+/**
+ * Collects information about the root package of a monorepo.
+ *
+ * @param args - The arguments to this function.
+ * @param args.packageDirectoryPath - The path to a package within a project.
+ * @param args.projectDirectoryPath - The path to the project directory.
+ * @param args.projectTagNames - The tag names across the whole project.
  * @returns Information about the package.
  */
-export async function readPackage(
-  packageDirectoryPath: string,
-): Promise<Package> {
+export async function readMonorepoRootPackage({
+  packageDirectoryPath,
+  projectDirectoryPath,
+  projectTagNames,
+}: {
+  packageDirectoryPath: string;
+  projectDirectoryPath: string;
+  projectTagNames: string[];
+}): Promise<Package> {
   const manifestPath = path.join(packageDirectoryPath, MANIFEST_FILE_NAME);
   const changelogPath = path.join(packageDirectoryPath, CHANGELOG_FILE_NAME);
   const { unvalidated: unvalidatedManifest, validated: validatedManifest } =
     await readPackageManifest(manifestPath);
+  const expectedReleaseTagNames = generateMonorepoRootPackageReleaseTagNames(
+    validatedManifest.version.toString(),
+  );
+  const tagNameMatchingLatestRelease = expectedReleaseTagNames.find(
+    (tagName) => {
+      return projectTagNames.includes(tagName);
+    },
+  );
+  // TODO: What if tags exist (projectTagNames.length > 0) but there is no tag
```

Comment:
@Gudahtt What do you think would be a good way to handle this?

Reply:
Hmm. Good question. Maybe it would be best to throw an error in this case? It would mean either that:

* The repository is in an invalid state (missing a tag), which would have to be fixed manually
* Someone is using this tool incorrectly (e.g. manually bumping the manifest version)
* Someone is running this in a repo that doesn't have any tags (e.g. they checked out a commit directly rather than the whole repo).

In all of those cases, we would not want to proceed

Reply:
Well, one other case where this would happen is on first run. We do want it to work then. Hmmm.

For the auto-changelog CLI, we assumed that the package had changed if we couldn't find the most recent tag. Maybe that's fine here as well.

Reply:
Maybe we could add a flag at some point to skip tag validation, for if this tool is being first used in a project that used different conventions previously

Reply:
Or we could ask the user for confirmation before proceeding

Reply:
> Maybe we could add a flag at some point to skip tag validation, for if this tool is being first used in a project that used different conventions previously

That's a good point. The only new tags we are introducing are tags for monorepo workspace packages. For existing projects, we won't have these tags, and they will only appear after the first release. I have (hopefully) accounted for this by checking for a tag to exist that's formatted like "<package name>@<version>", but *also* checking for a tag that corresponds to the root package, as a fallback.

Reply:
Updated here: [4986554](https://github.com/MetaMask/create-release-branch/pull/23/commits/498655440ed6433c5731ad7cb52879ee0b9b7dad)